### PR TITLE
Total refactor + update of extension 

### PR DESCRIPTION
**Pull request by NoahStapp**

### CODE REVIEW

**Comment by ellisonbg** [comment]
@NoahStapp great work! @gnestor can you have a look at this?

File: src/components/FileItem.tsx [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+  extractFilename: Function
+  contextMenu: Function
+  parseFileExtension: Function
+  selectedFile: number
+  updateSelectedFile: Function
+  fileIndex: number
+} 
+
+export class FileItem extends React.Component<IFileItemProps, {}> {
+  getFileChangedLabel(change: string) : string {
```
Comment (gnestor) [inline src/components/FileItem.tsx]: This is a personal style preference, but if a method is pure (doesn't depend on `this`), I like hoist it out of the component class (e.g. move it out of the class and into it's own function).

File: src/components/FileList.tsx [hunk 2]
```diff
@@ -0,0 +1,574 @@
...
+        buttons: [Dialog.warnButton({ label: 'OK' })]
+      }
+    ).then(result => {
+      if (result.button.accept) {
+        return
+      }
+    })
+   return
+  } try {
+    let leftSidebarItems = app.shell.widgets('left')
```
Comment (gnestor) [inline src/components/FileList.tsx]: Another personal preference: use `const` unless you *need* to reassign a variable

**Comment by Ennosigaeon** [comment]
Is there a reason why the pull and push functionality was removed?

**Pull request merged**